tests/internal/aws_client_mock.c

```diff
@@ -0,0 +1,250 @@
+            /* Search for header in request */
+            for (int h = 0; h < dynamic_headers_len; ++h) {
+                if (strncmp(dynamic_headers[h].key, (char *)val1,
+                            dynamic_headers[h].key_len) == 0) {
+                    /* Check header value */
+                    TEST_CHECK(strncmp(dynamic_headers[h].val, (char *)val2,
+                                       dynamic_headers[h].val_len + 1) == 0);
+                    TEST_MSG("[aws_mock_client] Expected Header: (%s: %s)", (char *)val1,
+                             (char *)val2);
+                    TEST_MSG("[aws_mock_client] Received Header: (%s: %s)", (char *)val1,
+                             dynamic_headers[h].val);
+
+                    header_found = FLB_TRUE;
+                }
+            }
+            TEST_CHECK(header_found);
+            TEST_MSG("[aws_mock_client] Expected Header: (%s: %s)", (char *)val1,
+                     (char *)val2);
+            TEST_MSG("[aws_mock_client] Header not received");
+        }
+        else if (response_config->config_parameter == FLB_AWS_CLIENT_MOCK_EXPECT_METHOD) {
+            char *flb_http_methods[] = {
+                "FLB_HTTP_GET",  "FLB_HTTP_POST",    "FLB_HTTP_PUT",
+                "FLB_HTTP_HEAD", "FLB_HTTP_CONNECT", "FLB_HTTP_PATCH",
+            };
+
+            /*
+             * Check method is what is expected
+             * Typecast config value from void * -> int
+             */
+            TEST_CHECK(method == (int)(uintptr_t)val1);
+            TEST_MSG("[aws_mock_client] Expected HTTP Method: %s",
+                     flb_http_methods[(int)(uintptr_t)val1]);
+            TEST_MSG("[aws_mock_client] Received HTTP Method: %s",
+                     flb_http_methods[method]);
+        }
+        else if (response_config->config_parameter ==
+                 FLB_AWS_CLIENT_MOCK_EXPECT_HEADER_COUNT) {
+            TEST_CHECK(dynamic_headers_len == (int)(uintptr_t)val1);
+            TEST_MSG("[aws_mock_client] Expected %d Headers", (int)(uintptr_t)val1);
+            TEST_MSG("[aws_mock_client] Received %d Headers",
+                     (int)(uintptr_t)dynamic_headers_len);
+        }
+        else if (response_config->config_parameter == FLB_AWS_CLIENT_MOCK_EXPECT_URI) {
+            TEST_CHECK(strncmp(uri, (char *)val1, strlen((char *)val1) + 1) == 0);
+            TEST_MSG("[aws_mock_client] Expected URI: %s", (char *)val1);
+            TEST_MSG("[aws_mock_client] Received URI: %s", uri);
+        }
+
+        /* Replace response client */
+        else if (response_config->config_parameter ==
+                 FLB_AWS_CLIENT_MOCK_CONFIG_REPLACE) {
+            flb_http_client_destroy(c);
+            c = (struct flb_http_client *)val1;
+        }
+
+        /*
+        * Response setters
+        * Set client fields using XMacro definitions
+        */
+#define EXPAND_CLIENT_RESPONSE_PARAMETER(lower, UPPER, type)                           \
```

Comment:
I think these macros have to always be prefixed with FLB

Reply:
Not sure if this is applicable, but lmk. This is actually following an XMacro pattern. So EXPAND_CLIENT_RESPONSE_PARAMETER definition is short lived and not accessible outside of the following one line of code. It defines a pattern which is copy pasted by `#include "aws_client_mock_client_resp.def"` for each of the listed uses. The definition however is removed just after the include, see `#undef EXPAND_CLIENT_RESPONSE_PARAMETER` so I didn't think it would be appropriate to give it a FLB_ prefix.

Reply:
Ok- if its not accessible outside of this file then its fine.